src/jvmMain/kotlin/io/github/fourlastor/wilds_launcher/ui/Launcher.kt

```diff
@@ -1,67 +1,166 @@
+                    }
+                }
+            }
+
+            Row(horizontalArrangement = Arrangement.End, modifier = Modifier.fillMaxWidth()) {
+                Button({ clearSettings() }, colors = ButtonDefaults.buttonColors(backgroundColor = Color.LightGray, contentColor = Color.Red)) {
+                    Text(text = "Clear Settings")
+                }
 
-        Text(settingsState.logs)
+                Spacer(modifier = Modifier.width(Dp(10f)))
+
+                Button(checkForUpdates) {
+                    Text(text = "Check for Updates")
+                }
+
+                Spacer(modifier = Modifier.width(Dp(10f)))
+
+                if (isPokeWildsStarted) {
+                    Button({
+                        context.process!!.destroy()
+                        context.process = null
+                    }, colors = ButtonDefaults.buttonColors(Color.Red)) {
+                        Text("Stop PokeWilds")
+                    }
+                }
+                else {
+                    Button({
+                        val threadContext = newSingleThreadContext("pokeWildsJar")
+
+                        context.coroutineScope.launch(threadContext) {
+                            var log: ((String) -> Unit)? = null
+
+                            if (context.settingsService.getLogsEnabled()) {
+                                log = { context.logger.log(it) }
+                            }
+
+                            val file = File(context.settingsService.getJar())
+
+                            val runArgs = mutableListOf("java", "-jar", file.absolutePath).apply {
+                                if (context.settingsService.getAngleGles20()) {
+                                    add("angle_gles20")
+                                }
+
+                                if (context.settingsService.getDevMode()) {
+                                    add("dev")
+                                }
+                            }.toTypedArray()
+
+                            try {
+                                log?.invoke("Running ${runArgs.joinToString(" ")}")
+
+                                val process = withContext(Dispatchers.IO) {
+                                    ProcessBuilder(*runArgs).directory(file.parentFile).start()
+                                }
+
+                                if (log != null) {
+                                    process.inputStream.capture(log)
+                                    process.errorStream.capture(log)
+                                }
+
+                                context.process = process
+
+                                isPokeWildsStarted = true
+                                process.onExit().thenApply { isPokeWildsStarted = false }
+                            } catch (exception: Throwable) {
+                                log?.invoke(exception.fullTrace())
+                            }
+                        }
```

Comment:
this should be extracted in a service/class

Reply:
Yup, fully agree! Hacked it in there